project_wbs/models/account_analytic_account.py

```diff
@@ -0,0 +1,219 @@
+            acc = account
+            while acc:
+                if acc.name and acc.parent_id:
+                    data.insert(0, '>')
+
+                acc = acc.parent_id
+            if data:
+                if len(data) >= 2:
+                    data = ''.join(data)  # pragma: no cover
+                else:
+                    data = data[0]
+            account.wbs_indent = data or ''
+
+    @api.multi
+    @api.depends('account_class', 'parent_id')
+    def _compute_project_analytic_id(self):
+        for analytic in self:
+            current = analytic
+            while current:
+                if current.account_class == 'project':
+                    analytic.project_analytic_id = current.id
+                    break
+                current = current.parent_id
+
+    @api.model
+    def _default_parent(self):
+        return self.env.context.get('parent_id', None)
+
+    @api.model
+    def _default_partner(self):
+        return self.env.context.get('partner_id', None)
+
+    @api.model
+    def _default_user(self):
+        return self.env.context.get('user_id', self.env.user)
+
+    wbs_indent = fields.Char(
+        compute=_wbs_indent_calc,
+        string='Level'
+    )
+
+    complete_wbs_code = fields.Char(
+        compute=_complete_wbs_code_calc,
+        string='Full WBS Code',
+        help='The full WBS code describes the full path of this component '
+             'within the project WBS hierarchy',
+        store=True,
+    )
+    code = fields.Char(copy=False)
+    complete_wbs_name = fields.Char(
+        compute=_complete_wbs_name_calc,
+        string='Full WBS path',
+        help='Full path in the WBS hierarchy',
+        store=True
+    )
+    project_analytic_id = fields.Many2one(
+        'account.analytic.account',
+        compute=_compute_project_analytic_id,
+        string='Root Analytic Account',
+        store=True
+    )
+    user_id = fields.Many2one(
+        'res.users', 'Project Manager', track_visibility='onchange',
+        default=_default_user)
+    manager_id = fields.Many2one('res.users', 'Manager',
+                                 track_visibility='onchange')
+
+    account_class = fields.Selection(
+        [('project', 'Project'), ('phase', 'Phase'),
+         ('deliverable', 'Deliverable'),
+         ('work_package', 'Work Package')], 'Class', default='project',
+        help='The classification allows you to create a proper project '
+             'Work Breakdown Structure'
+    )
+    parent_id = fields.Many2one(default=_default_parent,
+                                string="Parent Analytic Account")
+    partner_id = fields.Many2one(default=_default_partner)
+
+    @api.multi
+    def copy(self, default=None):
+        if self.mapped('project_ids'):
+            raise ValidationError(_(' Duplicate the project instead of the '
+                                    ' Analytic Account '))
+        default = {}
+        default['code'] = self.env['ir.sequence'].next_by_code(
```

Comment:
This execution branch is not covered with tests, please cover it as well

Reply:
IMO now it is covered